services/transcode/preview.go

```diff
@@ -0,0 +1,93 @@
+package transcode
+
+import (
+	"fmt"
+	"github.com/bcc-code/bccm-flows/utils"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"strings"
+)
+
+type PreviewInput struct {
+	FilePath  string
+	OutputDir string
+}
+
+type PreviewResult struct {
+	LowResolutionPath string
+}
+
+var previewWatermarkPath = os.Getenv("PREVIEW_WATERMARK_PATH")
+
+func Preview(input PreviewInput, progressCallback func(float64)) (*PreviewResult, error) {
```

Comment:
How does this react if it gets fed a audio file?
Here is a sample of what I used to encode audio only tracks:

```
ffmpeg -hide_banner -loglevel +level -y -ss 0.0 -i /path/to/filename.wav -filter_complex sws_flags=bicubic;[0:a:0]asplit=1[AUDIO-main-.mp4-0];[AUDIO-main-.mp4-0]aformat=channel_layouts=stereo[AUDIO-.mp4-0] -map [AUDIO-.mp4-0] -vn -c:a:0 aac -ar:a:0 44100 -b:a:0 128k /path/to/result.mp4
```